force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -192,13 +247,23 @@ public with sharing class ClassesObjectsMethods {
     public static String getDayOfWeek(DateTime dt) {
         // Define a map of day of the week as a key and day number as the value
         // Monday = 1, Tuesday = 2, etc.
-
+        Map<String, Integer> dayOfWeekMap = new Map<String, Integer>{
+            'Monday' => 1,
+            'Tuesday' => 2,
+            'Wednesday' => 3,
+            'Thursday' => 4,
+            'Friday' => 5,
+            'Saturday' => 6,
+            'Sunday' => 7
+        };
         // Get the day name using .format('EEEE');
-
+        Date todayDate = Date.today();
+        DateTime date1 = DateTime.newInstance(todayDate, Time.newInstance(0, 0, 0, 0));
+        String dayOfWeek = date1.format('EEEE');
```

Comment:
That's great that you were able to get this one. Try to see if you can condense the code a little bit by declaring the string variable and doing the format on the 'dt' that's been passed in, that way you can remove a couple of lines of code.